examples/DotNetCore/ConsoleAppWithBlobStorageReferences/Program.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Azure.Data.AppConfiguration;
+using Azure.Identity;
+using Azure.Storage.Blobs;
+using System.Text;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Examples.ConsoleApplication
+{
+    /*
+        Example Blob Content
+        -------------------------
+
+        {
+            "Data": [
+                "Lorem",
+                "ipsum",
+                "dolor",
+                "sit",
+                "amet"
+            ]
+        }
+    */
+
+    class MyBlobContent
+    {
+        public IList<string> Data { get; set; }
+    }
+
+    class Program
+    {
+        static IConfiguration Configuration { get; set; }
+        static IConfigurationRefresher _refresher;
+
+        static void Main(string[] args)
+        {
+            Configure();
+
+            var cts = new CancellationTokenSource();
+            _ = Run(cts.Token);
+
+            // Finish on key press
+            Console.ReadKey();
+            cts.Cancel();
+        }
+
+        private static async Task<string> ReadBlobContentAsync(Uri blobUri)
+        {
+            var blobClient = new BlobClient(blobUri, new DefaultAzureCredential());
+
+            // Read blob content
+            using (var contentStream = new MemoryStream())
+            {
+                await blobClient.DownloadToAsync(contentStream);
+
+                // Go to the beginning of the content stream
+                contentStream.Position = 0;
+
+                // Convert content to string
+                using (var reader = new StreamReader(contentStream))
+                {
+                    return reader.ReadToEnd();
+                }
+            }
+        }
+
+        private static void Configure()
+        {
+            var builder = new ConfigurationBuilder();
+
+            IConfiguration configuration = builder.Build();
+
+            if (string.IsNullOrEmpty(configuration["ConnectionString"]))
+            {
+                Console.WriteLine("Connection string not found.");
+                Console.WriteLine("Please set the 'ConnectionString' environment variable to a valid Azure App Configuration connection string and re-run this example.");
+                return;
+            }
+
+            // Augment the configuration builder with Azure App Configuration
+            // Pull the connection string from an environment variable
+            builder.AddAzureAppConfiguration(options =>
+            {
+                options.Connect(configuration["ConnectionString"])
+                        // key: BlobUri
+                        // value: https://{account_name}.blob.core.windows.net/{container_name}/{blob_name}
+                        .Select("BlobUri")
+                        .ConfigureRefresh(refresh =>
+                        {
+                            // Changes to BlobSentinel will refresh all key-values, including BlobUri
+                            refresh.Register("BlobSentinel", true);
```

Comment:
I feel like adding refresh in this example is making it more complicated and taking the focus away from Map API. Can you remove refresh and simplify the example? For example, we dont need a separate `Run` method and we can directly display blob content in main method.

Reply:
Yea it does look a bit complicated, I just assumed that the customer use case would most of the time involve a refresh. I can remove it and see how it looks.